packages/pipeline/src/data_sources/contract-wrappers/erc20_events.ts

```diff
@@ -0,0 +1,45 @@
+import {
+    ContractWrappers,
+    ERC20TokenApprovalEventArgs,
+    ERC20TokenEvents,
+    ERC20TokenWrapper,
+} from '@0x/contract-wrappers';
+import { Web3ProviderEngine } from '@0x/subproviders';
+import { LogWithDecodedArgs } from 'ethereum-types';
+
+import { GetEventsFunc, getEventsWithPaginationAsync } from './utils';
+
+export class ERC20EventsSource {
+    private readonly _erc20Wrapper: ERC20TokenWrapper;
+    private readonly _tokenAddress: string;
+    constructor(provider: Web3ProviderEngine, networkId: number, tokenAddress: string) {
+        const contractWrappers = new ContractWrappers(provider, { networkId });
+        this._erc20Wrapper = contractWrappers.erc20Token;
+        this._tokenAddress = tokenAddress;
+    }
+
+    public async getApprovalEventsAsync(
+        startBlock: number,
+        endBlock: number,
+    ): Promise<Array<LogWithDecodedArgs<ERC20TokenApprovalEventArgs>>> {
+        return getEventsWithPaginationAsync(
+            this._getApprovalEventsForRangeAsync.bind(this) as GetEventsFunc<ERC20TokenApprovalEventArgs>,
+            startBlock,
+            endBlock,
+        );
+    }
+
+    // Gets all approval events of for a specific sub-range. This getter
+    // function will be called during each step of pagination.
+    private async _getApprovalEventsForRangeAsync(
+        fromBlock: number,
```

Comment:
Did you mean to rename these variables "startBlock" and "endBlock"?

Reply:
Maybe this is not clear, but my intent was to distinguish between the large total range (`startBlock` to `endBlock`) and each individual sub-range within it (`fromBlock` to `toBlock`). Would a different naming scheme make it more clear?

Reply:
i think that makes sense, i just saw you changed many of the other names from fromBlock / ... to startBlock /... and wasn't sure if you meant to change these too.